### ISSUE TITLE
Find a proper way of supporting resumable uploads

### ISSUE THREAD

**Issue by michielbdejong** [remotestorage/spec#131]
We had to remove `Content-Range` headers on `PUT` requests in https://github.com/remotestorage/spec/pull/118 - so we will need to find a better solution for resumable uploads.

**Comment by fkooman** [comment]
Options: 
* The Google way (https://developers.google.com/drive/web/manage-uploads#resumable)
* http://www.resumablejs.com/
* http://plupload.com/

**Comment by untitaker** [comment]
The Google Way is what we removed, I think it can/should be ruled out.

The other libraries don't seem to talk about the underlying HTTP semantics at all.

**Comment by michielbdejong** [comment]
@untitaker the reason you don't like the way Google Drive does it is that it seems to violate the HTTP RFC to use `Content-Range` on `PUT`, right? Do you have a better idea? We could make it `X-Content-Range` :)

**Comment by untitaker** [comment]
Yes, just because it violates the RFC. I think it's tragic that this was explicitly disallowed, the way Google does it was very elegant until the new HTTP RFCs came along. Perhaps somebody from the IETF has a better suggestion?

**Comment by skddc** [comment]
@untitaker and I just had a nice long IRL discussion about what's missing for a 1.0 final of the RS spec. We agree that this is probably the most important missing feature at the moment.

Using an `X-Content-Range` header would work I think. If we apply for RFC status, this topic would also be reviewed by the IESG then. Or we can ask on an IETF mailing list beforehand.
